### ISSUE TITLE
Truncate syslog fields that are too long

### ISSUE THREAD

**Issue by jmervine** [heroku/log-iss#36]
Update log-iss to truncate incoming event fields. The specific issue is that we're getting an `APP-NAME` (per syslog spec[1]) that is longer than the max allowed length (48 characters). Ideally, we'd truncate all fields to the syslog field max length spec[1]. Because this project is open source and this is (probably) a Heroku only issue, we should include a configuration flag to enable / disable truncation.

TL;DR:
- [ ] Truncate the following syslog-ng fields, per the spec[1]:
  - [ ] `APP-NAME`
  - [ ] `PROCID`
  - [ ] `MSGID`
  - [ ] `PRIVAL`
  - [ ] `VERSION`
  - [ ] `HOSTNAME` 
- [ ] Add configuration flag for enabling / disabling truncation.
- [ ] Emit a counter metric for each field truncated.

[1] https://tools.ietf.org/html/rfc5424#section-6

**Comment by jmervine** [comment]
@kgarigipati @sheax0r or I can pair with you on this as necessary.

**Comment by sheax0r** [comment]
@jmervine @kgarigipati ping -- I think we should try to schedule the work on this, or decide that we're not going to do it soon.

**Comment by kgarigipati** [comment]
@sheax0r I did started working on this looongback, but haven't got chance to complete it.

I would like it put it in backlog, if thats fine, so that we wont forget

**Comment by jmervine** [comment]
Moved to GUS -- https://gus.my.salesforce.com/a07B0000006R8qkIAC
